### PR TITLE
#28536 Add more unit tests for IndexReviseEngine

### LINKED ISSUE #28536: Add more unit tests for IndexReviseEngine
Hi community,

This issue is for https://github.com/apache/shardingsphere/discussions/28487.

## Aim
Add more unit tests for IndexReviseEngine to test its public functions to improve unit test coverage.

Unit test coverage can be viewed at: https://app.codecov.io/gh/apache/shardingsphere

## Basic Qualifications
- Java
- Maven
- Junit

### PR DESCRIPTION
**Pull request by JessikaFujimura**
Fixes #28536.

Changes proposed in this pull request:
  - Add Unit Test for class IndexReviseEngine.java

---

Before committing this PR, I'm sure that I have checked the following options:
- [x] My code follows the [code of conduct](https://shardingsphere.apache.org/community/en/involved/conduct/code/) of this project.
- [x] I have self-reviewed the commit code.
- [x] I have (or in comment I request) added corresponding labels for the pull request.
- [x] I have passed maven check locally : `./mvnw clean install -B -T1C -Dmaven.javadoc.skip -Dmaven.jacoco.skip -e`.
- [ ] I have made corresponding changes to the documentation. (N/A)
- [ ] I have added corresponding unit tests for my changes. (N/A)

### CODE REVIEW

File: infra/common/src/test/java/org/apache/shardingsphere/infra/metadata/database/schema/reviser/index/IndexReviseEngineTest.java [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    
+    @BeforeEach
+    void setUp() {
+        MockitoAnnotations.openMocks(this);
+    }
+    
+    @Test
+    void assertReviseIsPresentIsFalse() {
+        when(metaDataReviseEntry.getIndexReviser(any(), anyString())).thenReturn(Optional.empty());
+
```
Comment (tuichenchuxin) [inline infra/common/src/test/java/org/apache/shardingsphere/infra/metadata/database/schema/reviser/index/IndexReviseEngineTest.java:66]: please remove empty line

File: infra/common/src/test/java/org/apache/shardingsphere/infra/metadata/database/schema/reviser/index/IndexReviseEngineTest.java [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+        MockitoAnnotations.openMocks(this);
+    }
+    
+    @Test
+    void assertReviseIsPresentIsFalse() {
+        when(metaDataReviseEntry.getIndexReviser(any(), anyString())).thenReturn(Optional.empty());
+        
+        Collection<IndexMetaData> indexMetaDataCollection = Collections.singletonList(new IndexMetaData("index"));
+        Collection<IndexMetaData> actual = indexReviseEngine.revise("tableName", indexMetaDataCollection);
+
```
Comment (tuichenchuxin) [inline infra/common/src/test/java/org/apache/shardingsphere/infra/metadata/database/schema/reviser/index/IndexReviseEngineTest.java:69]: please remove empty line

File: infra/common/src/test/java/org/apache/shardingsphere/infra/metadata/database/schema/reviser/index/IndexReviseEngineTest.java [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+        
+        Assertions.assertNotNull(actual);
+        assertThat(actual.size(), is(1));
+        assertThat(actual, equalToObject(indexMetaDataCollection));
+    }
+    
+    @Test
+    void assertReviseIsPresentIsTrue() {
+        IndexReviser<T> reviser = mock(IndexReviser.class);
+
```
Comment (tuichenchuxin) [inline infra/common/src/test/java/org/apache/shardingsphere/infra/metadata/database/schema/reviser/index/IndexReviseEngineTest.java:78]: please remove empty line

File: infra/common/src/test/java/org/apache/shardingsphere/infra/metadata/database/schema/reviser/index/IndexReviseEngineTest.java [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+        assertThat(actual.size(), is(1));
+        assertThat(actual, equalToObject(indexMetaDataCollection));
+    }
+    
+    @Test
+    void assertReviseIsPresentIsTrue() {
+        IndexReviser<T> reviser = mock(IndexReviser.class);
+        
+        IndexMetaData indexMetaData = new IndexMetaData("index");
+
```
Comment (tuichenchuxin) [inline infra/common/src/test/java/org/apache/shardingsphere/infra/metadata/database/schema/reviser/index/IndexReviseEngineTest.java:80]: please remove empty line

File: infra/common/src/test/java/org/apache/shardingsphere/infra/metadata/database/schema/reviser/index/IndexReviseEngineTest.java [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+    void assertReviseIsPresentIsTrue() {
+        IndexReviser<T> reviser = mock(IndexReviser.class);
+        
+        IndexMetaData indexMetaData = new IndexMetaData("index");
+        
+        doReturn(Optional.of(reviser)).when(metaDataReviseEntry).getIndexReviser(any(), anyString());
+        when(reviser.revise(anyString(), any(), any())).thenReturn(Optional.of(indexMetaData));
+        
+        Collection<IndexMetaData> indexMetaDataCollection = Arrays.asList(new IndexMetaData("index1"), new IndexMetaData("index2"));
+
```
Comment (tuichenchuxin) [inline infra/common/src/test/java/org/apache/shardingsphere/infra/metadata/database/schema/reviser/index/IndexReviseEngineTest.java:85]: please remove empty line

File: infra/common/src/test/java/org/apache/shardingsphere/infra/metadata/database/schema/reviser/index/IndexReviseEngineTest.java [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+        
+        IndexMetaData indexMetaData = new IndexMetaData("index");
+        
+        doReturn(Optional.of(reviser)).when(metaDataReviseEntry).getIndexReviser(any(), anyString());
+        when(reviser.revise(anyString(), any(), any())).thenReturn(Optional.of(indexMetaData));
+        
+        Collection<IndexMetaData> indexMetaDataCollection = Arrays.asList(new IndexMetaData("index1"), new IndexMetaData("index2"));
+        
+        Collection<IndexMetaData> actual = indexReviseEngine.revise("tableName", indexMetaDataCollection);
+
```
Comment (tuichenchuxin) [inline infra/common/src/test/java/org/apache/shardingsphere/infra/metadata/database/schema/reviser/index/IndexReviseEngineTest.java:87]: please remove empty line

**Review by tuichenchuxin — CHANGES_REQUESTED** [review]

File: infra/common/src/test/java/org/apache/shardingsphere/infra/metadata/database/schema/reviser/index/IndexReviseEngineTest.java [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.when;
+
+public class IndexReviseEngineTest<T extends ShardingSphereRule> {
```
Comment (tuichenchuxin) [inline infra/common/src/test/java/org/apache/shardingsphere/infra/metadata/database/schema/reviser/index/IndexReviseEngineTest.java:47]: please remove public

File: infra/common/src/test/java/org/apache/shardingsphere/infra/metadata/database/schema/reviser/index/IndexReviseEngineTest.java (hunk 1 above)
Comment (JessikaFujimura) [reply infra/common/src/test/java/org/apache/shardingsphere/infra/metadata/database/schema/reviser/index/IndexReviseEngineTest.java]: Done! Thanks :)

File: infra/common/src/test/java/org/apache/shardingsphere/infra/metadata/database/schema/reviser/index/IndexReviseEngineTest.java (hunk 2 above)
Comment (JessikaFujimura) [reply infra/common/src/test/java/org/apache/shardingsphere/infra/metadata/database/schema/reviser/index/IndexReviseEngineTest.java]: Done! Thanks :)

File: infra/common/src/test/java/org/apache/shardingsphere/infra/metadata/database/schema/reviser/index/IndexReviseEngineTest.java (hunk 3 above)
Comment (JessikaFujimura) [reply infra/common/src/test/java/org/apache/shardingsphere/infra/metadata/database/schema/reviser/index/IndexReviseEngineTest.java]: Done! Thanks :)

File: infra/common/src/test/java/org/apache/shardingsphere/infra/metadata/database/schema/reviser/index/IndexReviseEngineTest.java (hunk 4 above)
Comment (JessikaFujimura) [reply infra/common/src/test/java/org/apache/shardingsphere/infra/metadata/database/schema/reviser/index/IndexReviseEngineTest.java]: Done! Thanks :)

File: infra/common/src/test/java/org/apache/shardingsphere/infra/metadata/database/schema/reviser/index/IndexReviseEngineTest.java (hunk 5 above)
Comment (JessikaFujimura) [reply infra/common/src/test/java/org/apache/shardingsphere/infra/metadata/database/schema/reviser/index/IndexReviseEngineTest.java]: Done! Thanks :)

File: infra/common/src/test/java/org/apache/shardingsphere/infra/metadata/database/schema/reviser/index/IndexReviseEngineTest.java (hunk 6 above)
Comment (JessikaFujimura) [reply infra/common/src/test/java/org/apache/shardingsphere/infra/metadata/database/schema/reviser/index/IndexReviseEngineTest.java]: Done! Thanks :)

File: infra/common/src/test/java/org/apache/shardingsphere/infra/metadata/database/schema/reviser/index/IndexReviseEngineTest.java (hunk 7 above)
Comment (JessikaFujimura) [reply infra/common/src/test/java/org/apache/shardingsphere/infra/metadata/database/schema/reviser/index/IndexReviseEngineTest.java]: Done! Thanks :)

**Review by tuichenchuxin — APPROVED** [review]

**Pull request merged**